src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Components/ComponentInjectIntermediateNode.cs

```diff
@@ -59,13 +62,18 @@ public override void WriteNode(CodeTarget target, CodeRenderingContext context)
             throw new ArgumentNullException(nameof(context));
         }
 
-        context.CodeWriter.WriteAutoPropertyDeclaration(
-            _injectedPropertyModifiers,
-            TypeName,
-            MemberName,
-            TypeSpan,
-            MemberSpan,
-            context,
-            defaultValue: true);
+        var memberName = MemberName ?? "Member_" + DefaultTagHelperTargetExtension.GetDeterministicId(context);
+
+        if (!context.Options.DesignTime || !IsMalformed)
```

Comment:
Will it mess up the existing design-time experience if we just unconditionally write this property?